geonode/geoserver/security.py

```diff
@@ -447,15 +414,36 @@ def has_geolimits(layer, user, group):
 
 
 def _get_gf_services(layer, perms):
-    gf_services = {}
-    gf_services["WMS"] = "view_resourcebase" in perms or "change_dataset_style" in perms
-    gf_services["GWC"] = "view_resourcebase" in perms or "change_dataset_style" in perms
-    gf_services["WFS"] = ("download_resourcebase" in perms or "change_dataset_data" in perms) and layer.is_vector()
-    gf_services["WCS"] = ("download_resourcebase" in perms or "change_dataset_data" in perms) and not layer.is_vector()
-    gf_services["WPS"] = "download_resourcebase" in perms or "change_dataset_data" in perms
-    gf_services["*"] = "download_resourcebase" in perms and (
-        "view_resourcebase" in perms or "change_dataset_style" in perms
-    )
+    edit = "change_dataset_data" in perms
+    download = "download_resourcebase" in perms
+    view = "view_resourcebase" in perms
+
+    gf_services = []
+
+    # view services
+    if view or "change_dataset_style" in perms:
+        gf_services.append({"service": "WMS", "access": True})
+        gf_services.append({"service": "GWC", "access": True})
+
+    # WPS
+    if view or download or edit:
+        if not download:
+            ver = gs_catalog.get_version()
+            if version.parse(ver) >= version.parse("2.23.0"):
```

Comment:
@etj version.parse throws an exception in case of "2.23.SNAPSHOT". I know we should always have a released version for Geoserver, but we all know that sometime it happens that we have to use a SNAPSHOT in edge cases.
We should probably use a more relaxed version parsing solution.

Reply:
Uhm, the library for parsing the version is `packaging`, but it requires the format `[N!]N(.N)*[{a|b|rc}N][.postN][.devN]`. I guess we need to perform some parsing by hand.